dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFile.h

```diff
@@ -114,7 +114,8 @@ class ColumnFile
     virtual ColumnFileReaderPtr
     getReader(const DMContext & context, const StorageSnapshotPtr & storage_snap, const ColumnDefinesPtr & col_defs) const = 0;
 
-    /// only ColumnInMemoryFile can be appendable
+    /// Note: Only ColumnFileInMemory can be appendable. Other ColumnFiles (i.e. ColumnFilePersisted) have
+    /// already been persisted in the disk and they should be immutable.
```

Comment:
For `ColumnFileTiny`, the data is mutable, but the column file object is still mutable. For example, we can clear it's cache.